js/languagebox.js

```diff
@@ -12,151 +12,152 @@
 
 
 //A dropdown for selecting language
-var LANGUAGEBOX = '<svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="180" height="560"> <rect width="180" height="560" x="0" y="-3.5762787e-06" style="fill:#ffffff;fill-opacity:1;fill-rule:nonzero;stroke:none" /> <g transform="translate(124.943,-0.053)" style="fill:#000000;display:block"> <path d="m 27.557,5.053 c -12.43,0 -22.5,10.076 -22.5,22.497 0,12.432 10.07,22.503 22.5,22.503 12.431,0 22.5,-10.071 22.5,-22.503 0,-12.421 -10.07,-22.497 -22.5,-22.497 z m 10.199,28.159 c 1.254,1.256 1.257,3.291 0,4.545 -0.628,0.629 -1.451,0.943 -2.274,0.943 -0.822,0 -1.644,-0.314 -2.27,-0.94 l -5.76,-5.761 -5.76,5.761 c -0.627,0.626 -1.449,0.94 -2.271,0.94 -0.823,0 -1.647,-0.314 -2.275,-0.943 -1.254,-1.254 -1.254,-3.289 0.004,-4.545 l 5.758,-5.758 -5.758,-5.758 c -1.258,-1.254 -1.258,-3.292 -0.004,-4.546 1.255,-1.254 3.292,-1.259 4.546,0 l 5.76,5.759 5.76,-5.759 c 1.252,-1.259 3.288,-1.254 4.544,0 1.257,1.254 1.254,3.292 0,4.546 l -5.758,5.758 5.758,5.758 z" style="fill:#000000;display:inline" /> </g> <rect width="174" height="505" x="3" y="57" style="fill:#92b5c8;fill-opacity:1;stroke:none" /> </svg>';
-var language = localStorage.languagePreference;
-this._message = null;
-this._container = null;
-var myDiv = docById('languageDiv');
-this._scale = 1;
-
-this.setMessage = function (message) {
-    this._message = message;
-    return this;
-};
-
-this.enUS_onclick=function(){
-    language='en-US';
-};
-
-this.enUK_onclick=function(){
-    language='en-UK';
-};
-
-this.ja_onclick=function(){
-    language='ja';
-};
-
-this.kana_onclick=function(){
-    language='kana';
-};
-
-this.es_onclick=function(){
-    language='es';
-};
-
-this.zhCN_onclick=function(){
-    language='zh-CN';
-};
-
-this.th_onclick=function(){
-    language='th';
-};
-
-this.hi_onclick=function(){
-    language='hi';
-};
-
-this.ibo_onclick=function(){
-    language='ibo';
-};
-
-this.ar_onclick=function(){
-    language='ar';
-};
-
-this.he_onclick=function(){
-    language='he';
-};
-
-this.ayc_onclick=function(){
-    language='ayc';
-};
-
-this.gug_onclick=function(){
-    language='gug';
-};
+function LanguageBox () {
+    var language = localStorage.languagePreference;
+    this._message = null;
+    this._container = null;
+    var myDiv = docById('languageDiv');
+    this._scale = 1;
+
+    this.setMessage = function (message) {
+        this._message = message;
+        return this;
+    };
+
+    this.enUS_onclick=function(){
```

Comment:
inside each of these, call this.hide() after setting the language so the message is displayed and the language selection is stored.

Reply:
Actually, you need to store the language selection here.